service/src/test/java/bio/terra/workspace/service/workspace/AzureWorkspaceTest.java

```diff
@@ -0,0 +1,64 @@
+package bio.terra.workspace.service.workspace;
+
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import bio.terra.workspace.app.configuration.external.AzureTestConfiguration;
+import bio.terra.workspace.common.BaseAzureTest;
+import bio.terra.workspace.connected.WorkspaceConnectedTestUtils;
+import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import bio.terra.workspace.service.job.JobService;
+import bio.terra.workspace.service.spendprofile.SpendConnectedTestUtils;
+import bio.terra.workspace.service.workspace.model.AzureCloudContext;
+import bio.terra.workspace.service.workspace.model.Workspace;
+import bio.terra.workspace.service.workspace.model.WorkspaceStage;
+import java.util.Optional;
+import java.util.UUID;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class AzureWorkspaceTest extends BaseAzureTest {
+
+  @Autowired private AzureTestConfiguration azureTestConfiguration;
+  @Autowired private JobService jobService;
+  @Autowired private SpendConnectedTestUtils spendUtils;
+  @Autowired private WorkspaceConnectedTestUtils testUtils;
+  @Autowired private WorkspaceService workspaceService;
+
+  @Test
+  void createGetDeleteAzureContext() {
+    AuthenticatedUserRequest userRequest =
+        new AuthenticatedUserRequest()
+            .token(Optional.of("fake-token"))
+            .email("fake@email.com")
+            .subjectId("fakeID123");
+
+    Workspace request =
+        Workspace.builder()
+            .workspaceId(UUID.randomUUID())
+            .spendProfileId(spendUtils.defaultSpendId())
+            .workspaceStage(WorkspaceStage.MC_WORKSPACE)
+            .build();
+
+    workspaceService.createWorkspace(request, userRequest);
+
+    String jobId = UUID.randomUUID().toString();
+    AzureCloudContext azureCloudContext =
+        new AzureCloudContext(
+            azureTestConfiguration.getTenantId(),
+            azureTestConfiguration.getSubscriptionId(),
+            azureTestConfiguration.getManagedResourceGroupId());
+    workspaceService.createAzureCloudContext(
+        request.getWorkspaceId(), jobId, userRequest, "/fake/value", azureCloudContext);
+    jobService.waitForJob(jobId);
+
+    assertNull(jobService.retrieveJobResult(jobId, Object.class, userRequest).getException());
+    assertTrue(
+        testUtils
+            .getAuthorizedAzureCloudContext(request.getWorkspaceId(), userRequest)
+            .isPresent());
+    workspaceService.deleteAzureCloudContext(request.getWorkspaceId(), userRequest);
```

Comment:
Is it important to ensure that the cloud context is deleted? If so, this should be put into a finally block or test cleanup method.

Reply:
I don't think so. We're using a mock sam here so no external resources are leaked and we're not allocating any other resources as part of this test. In the event that we create real resources, we can implement an `@AfterAll` to cleanup anything remaining.